### DISCUSSION TITLE
Refresh interval causes Automation failure

### DISCUSSION THREAD

**Discussion by SGXander** [WillCodeForCats/solaredge-modbus-multi#706] [General]
### Describe the bug

I have the refresh interval set to 10s. When setting up an automation to run if I have it run exactly on a minute or 10s gap it fails with:
Error: Connection to inverter ID 1 failed.

### Expected behavior

The automation action should potentially queue behind the refresh so it takes effect immediately after.

### Screenshots

Broken on the minute, 10s past etc
![Screenshot_20241124_092821_Home Assistant](https://github.com/user-attachments/assets/dbc404a7-aee8-478f-8393-1f2183fb9337)

Working at 5s interval (also at 15, 25 etc)
![Screenshot_20241124_092836_Home Assistant](https://github.com/user-attachments/assets/fa299ae6-dfeb-47d3-b793-235398487b40)

### Diagnostic File

No diagnostic file taken

### Home Assistant Version

2024.11.3

### solaredge-modbus-multi Version

3.0.2

### Installation Type

HAOS

### Read the Instructions

- [X] My configuration follows https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Configuration
- [X] I have read the Known Issues https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Known-Issues
- [X] This issue is not about a template error (do not report template errors, read https://github.com/WillCodeForCats/solaredge-modbus-multi/wiki/Template-Design-Notes)

**Comment by WillCodeForCats** [comment]
Home Assistant core decides when refreshes are executed, not the integration.

**Comment by WillCodeForCats** [comment]
Someone else mentioned they use a retry service, you can try that:

[solaredge-modbus-multi/discussions/697](https://github.com/WillCodeForCats/solaredge-modbus-multi/discussions/697)

https://github.com/amitfin/retry
